Sources/Sandbox.Game/Game/Entities/MyEntity.cs

```diff
@@ -100,6 +100,13 @@ public MyGameLogicComponent GameLogic
             set { Components.Add<MyGameLogicComponent>(value); }
         }
 
+		private HashSet<MyGameLogicComponent> m_gameLogicSet = null;
```

Comment:
Why not just make m_gameLogic a set? Now the class has two fields that basically serve the same purpose.